### PR TITLE
[HBASE-22453] A NullPointerException could be thrown

### PR DESCRIPTION
**Pull request by q977734161**
The method incrementalRestoreTable(...) could be thrown  an NullPointerException and modify method checkAndCreateTable(...)'s doc

### CODE REVIEW

File: hbase-backup/src/main/java/org/apache/hadoop/hbase/backup/util/RestoreTool.java [hunk 1]
```diff
@@ -169,6 +169,9 @@ public void incrementalRestoreTable(Connection conn, Path tableBackupPath, Path[
       for (int i = 0; i < tableNames.length; i++) {
         TableName tableName = tableNames[i];
         TableDescriptor tableDescriptor = getTableDescriptor(fileSys, tableName, incrBackupId);
+        if (tableDescriptor == null) {
+          throw new IOException("Can't found " + tableName + "'s descriptor ");
```
Comment (pzygielo) [inline hbase-backup/src/main/java/org/apache/hadoop/hbase/backup/util/RestoreTool.java]: Can't <strike>found</strike>... :arrow_right: Can't find...

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 72 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | hbaseanti | 0 |  Patch does not have any anti-patterns. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -0 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ master Compile Tests _ |
| +1 | mvninstall | 255 | master passed |
| +1 | compile | 23 | master passed |
| +1 | checkstyle | 15 | master passed |
| +1 | shadedjars | 270 | branch has no errors when building our shaded downstream artifacts. |
| +1 | findbugs | 38 | master passed |
| +1 | javadoc | 16 | master passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 244 | the patch passed |
| +1 | compile | 25 | the patch passed |
| +1 | javac | 25 | the patch passed |
| +1 | checkstyle | 14 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedjars | 266 | patch has no errors when building our shaded downstream artifacts. |
| +1 | hadoopcheck | 953 | Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.0.3 3.1.2. |
| +1 | findbugs | 45 | the patch passed |
| +1 | javadoc | 16 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 1036 | hbase-backup in the patch passed. |
| +1 | asflicense | 13 | The patch does not generate ASF License warnings. |
| | | 3609 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-272/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/272 |
| JIRA Issue | HBASE-22453 |
| Optional Tests |  dupname  asflicense  javac  javadoc  unit  findbugs  shadedjars  hadoopcheck  hbaseanti  checkstyle  compile  |
| uname | Linux 30a721922307 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /testptch/patchprocess/precommit/personality/provided.sh |
| git revision | master / 2823090ce4 |
| maven | version: Apache Maven 3.5.4 (1edded0938998edf8bf061f1ceb3cfdeccf443fe; 2018-06-17T18:33:14Z) |
| Default Java | 1.8.0_181 |
| findbugs | v3.1.11 |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-272/1/testReport/ |
| Max. process+thread count | 4756 (vs. ulimit of 10000) |
| modules | C: hbase-backup U: hbase-backup |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-272/1/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

File: hbase-backup/src/main/java/org/apache/hadoop/hbase/backup/util/RestoreTool.java (hunk 1 above)
Comment (q977734161) [reply hbase-backup/src/main/java/org/apache/hadoop/hbase/backup/util/RestoreTool.java]: Oh,yes,some problems with grammar,I will modify it right away.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 43 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | hbaseanti | 0 |  Patch does not have any anti-patterns. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -0 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ master Compile Tests _ |
| +1 | mvninstall | 258 | master passed |
| +1 | compile | 23 | master passed |
| +1 | checkstyle | 14 | master passed |
| +1 | shadedjars | 268 | branch has no errors when building our shaded downstream artifacts. |
| +1 | findbugs | 39 | master passed |
| +1 | javadoc | 16 | master passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 253 | the patch passed |
| +1 | compile | 24 | the patch passed |
| +1 | javac | 24 | the patch passed |
| +1 | checkstyle | 14 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedjars | 278 | patch has no errors when building our shaded downstream artifacts. |
| +1 | hadoopcheck | 773 | Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2. |
| +1 | findbugs | 46 | the patch passed |
| +1 | javadoc | 16 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 1036 | hbase-backup in the patch passed. |
| +1 | asflicense | 15 | The patch does not generate ASF License warnings. |
| | | 3414 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-272/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/272 |
| JIRA Issue | HBASE-22453 |
| Optional Tests |  dupname  asflicense  javac  javadoc  unit  findbugs  shadedjars  hadoopcheck  hbaseanti  checkstyle  compile  |
| uname | Linux 3db89c169be8 4.4.0-137-generic #163-Ubuntu SMP Mon Sep 24 13:14:43 UTC 2018 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /testptch/patchprocess/precommit/personality/provided.sh |
| git revision | master / e4f4ae835f |
| maven | version: Apache Maven 3.5.4 (1edded0938998edf8bf061f1ceb3cfdeccf443fe; 2018-06-17T18:33:14Z) |
| Default Java | 1.8.0_181 |
| findbugs | v3.1.11 |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-272/2/testReport/ |
| Max. process+thread count | 4612 (vs. ulimit of 10000) |
| modules | C: hbase-backup U: hbase-backup |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-272/2/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 139 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | hbaseanti | 0 |  Patch does not have any anti-patterns. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -0 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ master Compile Tests _ |
| +1 | mvninstall | 264 | master passed |
| +1 | compile | 24 | master passed |
| +1 | checkstyle | 15 | master passed |
| +1 | shadedjars | 279 | branch has no errors when building our shaded downstream artifacts. |
| +1 | findbugs | 42 | master passed |
| +1 | javadoc | 17 | master passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 245 | the patch passed |
| +1 | compile | 23 | the patch passed |
| +1 | javac | 23 | the patch passed |
| +1 | checkstyle | 14 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedjars | 262 | patch has no errors when building our shaded downstream artifacts. |
| +1 | hadoopcheck | 764 | Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2. |
| +1 | findbugs | 47 | the patch passed |
| +1 | javadoc | 17 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 1025 | hbase-backup in the patch passed. |
| +1 | asflicense | 12 | The patch does not generate ASF License warnings. |
| | | 3517 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-272/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/272 |
| JIRA Issue | HBASE-22453 |
| Optional Tests |  dupname  asflicense  javac  javadoc  unit  findbugs  shadedjars  hadoopcheck  hbaseanti  checkstyle  compile  |
| uname | Linux 499134e6c5c9 4.4.0-139-generic #165-Ubuntu SMP Wed Oct 24 10:58:50 UTC 2018 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /testptch/patchprocess/precommit/personality/provided.sh |
| git revision | master / 26037854ad |
| maven | version: Apache Maven 3.5.4 (1edded0938998edf8bf061f1ceb3cfdeccf443fe; 2018-06-17T18:33:14Z) |
| Default Java | 1.8.0_181 |
| findbugs | v3.1.11 |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-272/3/testReport/ |
| Max. process+thread count | 4472 (vs. ulimit of 10000) |
| modules | C: hbase-backup U: hbase-backup |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-272/3/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

**Pull request merged**